### ISSUE TITLE
Improve drop indicator within the List view

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#39068]
## What problem does this address?
We currently have a bold indicator when dragging/dropping blocks within the editor, but in the List view, it is quite thin/not as easily distinguishable.

## What is your proposed solution?
1. Let's try a more consistent drop indicator here that also improves visibility of the indicator — perhaps 1.5 or even 2px wide. Perhaps even matching the focus state width (in the last screenshot). 

2. As an extra, perhaps lean into matching a corresponding animation (perhaps not as bold as the editor itself, but relative to the List view size) for the line indicator as it appears. 

#### Current drop indicator within the editor: 
<img width="667" alt="CleanShot 2022-02-24 at 13 01 28@2x" src="https://user-images.githubusercontent.com/1813435/155581238-89731555-5066-4326-835c-22f53a49abb1.png">

#### Current drop indicator within List view: 
<img width="341" alt="CleanShot 2022-02-24 at 13 02 1" src="https://user-images.githubusercontent.com/1813435/155581488-95e3c215-fedc-427c-b53d-7f6e9145fc83.png">
<img width="347" alt="CleanShot 2022-02-24 at 13 05 05@2x" src="https://user-images.githubusercontent.com/1813435/155581750-c4a1fd57-8d73-4c4b-b380-551b373a967a.png">

**Comment by talldan** [comment]
I noticed this disparity too recently. It'd be great to improve it. I don't think it'll be too difficult, once we have a design I think it can be considered a slightly more challenging 'Good First Issue'. 😄 

There has been some work on changing the way drag and drop works here - https://github.com/WordPress/gutenberg/pull/34022. But I think it'd also be fine to ship an interim improvement to the indicator.

For anyone interested in contributing, the code is here:
- List View drop indicator react component - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/list-view/drop-indicator.js
- List View drop indicator CSS - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/list-view/style.scss#L347-L361

**Comment by richtabor** [comment]
Probably worth a separate issue, but I've noticed that in Safari the drop indicator is not indicative of where the drop actually happens. 

https://user-images.githubusercontent.com/1813435/177634687-19087227-4783-4d6d-9f02-c2f3cd0d13b3.mp4

**Comment by talldan** [comment]
@richtabor Yep, there was an issue reported here for that - https://github.com/WordPress/gutenberg/issues/42058. It seems to be related to recent changes to the popover component.

**Comment by talldan** [comment]
This is a bit outdated now so I'll close it.
